src/rgw/store/simplefile/user.cc

```diff
@@ -0,0 +1,178 @@
+  return -ENOTSUP;
+}
+
+int SimpleFileUser::trim_usage(const DoutPrefixProvider *dpp,
+                               uint64_t start_epoch, uint64_t end_epoch) {
+  ldpp_dout(dpp, 10) << __func__ << ": TODO" << dendl;
+  return -ENOTSUP;
+}
+
+int SimpleFileUser::load_user(const DoutPrefixProvider *dpp, optional_yield y) {
+  ldpp_dout(dpp, 10) << __func__ << ": TODO (0)" << dendl;
+  return 0;
+}
+
+int SimpleFileUser::store_user(const DoutPrefixProvider *dpp, optional_yield y,
+                               bool exclusive, RGWUserInfo *old_info) {
+  /** Store this User to the backing store */ ldpp_dout(dpp, 10)
+      << __func__ << ": TODO" << dendl;
+  return -ENOTSUP;
+}
+
+int SimpleFileUser::remove_user(const DoutPrefixProvider *dpp,
+                                optional_yield y) {
+  ldpp_dout(dpp, 10) << __func__ << ": TODO" << dendl;
+  return -ENOTSUP;
+}
+
+static void populate_buckets_from_path(
+  SimpleFileStore *store,
+  const DoutPrefixProvider *dpp,
+  std::filesystem::path path,
+  BucketList& buckets
+) {
+  ldpp_dout(dpp, 10) << __func__ << ": from path " << path << dendl;
+  for (auto const &dir_entry : std::filesystem::directory_iterator{path}) {
+    auto p = dir_entry.path();
+    ldpp_dout(dpp, 10) << __func__ << ": bucket: " << p << dendl;
+    std::string bucketname = p.filename();
+    auto mgr = store->get_bucket_mgr(bucketname);
+    auto bucket =
+        std::unique_ptr<Bucket>(new SimpleFileBucket{p, store, mgr});
+    bucket->load_bucket(dpp, null_yield);
+    buckets.add(std::move(bucket));
+  }
+}
+
+int SimpleFileUser::list_buckets(const DoutPrefixProvider *dpp,
+                                 const std::string &marker,
+                                 const std::string &end_marker, uint64_t max,
+                                 bool need_stats, BucketList &buckets,
+                                 optional_yield y) {
+  // TODO this should list buckets assigned to a user. for now we just get every
+  // bucket
+  ldpp_dout(dpp, 10) << __func__
+                     << ": marker (" << marker << ", " << end_marker
+                     << "), max=" << max << dendl;
+  populate_buckets_from_path(store, dpp, store->buckets_path(), buckets);
+  ldpp_dout(dpp, 10) << __func__ << ": buckets=" << buckets.get_buckets()
+                     << dendl;
+  return 0;
+}
+
+int SimpleFileUser::create_bucket(
+    const DoutPrefixProvider *dpp,
+    const rgw_bucket &b,
+    const std::string &zonegroup_id,
+    rgw_placement_rule &placement_rule,
+    std::string &swift_ver_location,
+    const RGWQuotaInfo *pquota_info,
+    const RGWAccessControlPolicy &policy,
+    Attrs &attrs,
+    RGWBucketInfo &info,
+    obj_version &ep_objv,
+    bool exclusive,
+    bool obj_lock_enabled,
+    bool *existed,
+    req_info &req_info,
+    std::unique_ptr<Bucket> *bucket,
+    optional_yield y
+) {
+  ceph_assert(bucket != nullptr);
+
+  auto f = new JSONFormatter(true);
```

Comment:
Use `JSONFormatter f(true)` like https://github.com/aquarist-labs/ceph/pull/8/files#diff-2b9f89f3573abb0614d72a4533ce2140b34fb4222046482dcab3d848761d8da8R289 or ensure to delete the object after use.

Reply:
this needs to go away anyway. I don't think we can allow ourselves to write stuff as json anway.